hw/ip/aes/doc/_index.md

```diff
@@ -55,14 +59,14 @@ The benefits of this design compared to passing all round keys via register inte
   - one set of registers to hold the full key of the last encryption round, i.e., the start key for decryption.
 - Faster re-configuration and key switching: The core just needs to perform 8 write operations instead of 60 write operations for AES-256.
 
-On-the-fly round-key generation comes however at the price of an initial delay whenever the key is changed by the processor before the AES unit can perform ECB **decryption** using this new key.
+On-the-fly round-key generation comes however at the price of an initial delay whenever the key is changed by the processor before the AES unit can perform ECB/CBC **decryption** using this new key.
 During this phase, the key expanding mechanism iteratively computes the start key for the decryption.
 The duration of this delay phase corresponds to the latency required for encrypting one 16B block (i.e., 12/14/16 cycles for AES-128/192/256).
 Once the start key for decryption has been computed, it is stored in a dedicated internal register for later use.
 The AES unit can then switch between decryption and encryption without additional overhead.
 
-For encryption, there is no such initial delay upon changing the key.
-If the next operation after a key switch is ECB **decryption**, the AES unit automatically initiates an encryption procedure in the key expanding mechanism first (to generate the start key for decryption, the actual data path remains idle during that phase).
+For encryption or if the mode is set to CTR, there is no such initial delay upon changing the key.
+If the next operation after a key switch is ECB or CBC **decryption**, the AES unit automatically initiates an encryption procedure in the key expanding mechanism first (to generate the start key for decryption, the actual data path remains idle during that phase).
```

Comment:
This sentence is confusing (to me).  I think it should say "... the AES unit automatically initiates a key expansion using the key schedule [mechanism] first (to generate the first round key used for decryption, during that phase the cipher data path remains idle)" 

(up to you if "mechanism" should be in that sentence or not, I could see it either way.)

Reply:
I have switched to your suggestion.